### PR TITLE
Initial contract unit tests added

### PR DESCRIPTION
**Pull request by kevinshahfws**
## What

- [x] Pact contract test dependencies added with feature
- [x] Structured tests in separate test folder
- [x] Added macros for creating pact request/response map
- [x] Added around 30 initial contracts for rdk_service

### CODE REVIEW

File: device/thunder_ripple_sdk/src/tests/contracts/contract_utils.rs [hunk 1]
```diff
@@ -0,0 +1,143 @@
+use pact_consumer::mock_server::StartMockServerAsync;
```
Comment (satlead) [inline device/thunder_ripple_sdk/src/tests/contracts/contract_utils.rs:1]: Needs the License Header

**Review by satlead — CHANGES_REQUESTED** [review]
Very good start for Ripple's CI CD journey. Just minor changes

File: device/thunder_ripple_sdk/src/tests/contracts/contract_utils.rs (hunk 1 above)
Comment (kevinshahfws) [reply device/thunder_ripple_sdk/src/tests/contracts/contract_utils.rs:18]: Done

File: device/thunder_ripple_sdk/Cargo.toml [hunk 2]
```diff
@@ -23,10 +23,23 @@ edition = "2021"
...
+contract_tests = []
+
 [dependencies]
 ripple_sdk = { path = "../../core/sdk" }
 jsonrpsee = { version = "0.9.0", features = ["macros", "ws-client"]}
 serde = { version = "1.0", features = ["derive"] }
 url = "2.2.2"
 strum = "0.24"
 strum_macros = "0.24"
+tokio = "1.16.1"
```
[the remark this answers is not in the document]
Comment (kevinshahfws) [reply device/thunder_ripple_sdk/Cargo.toml]: Done

File: device/thunder_ripple_sdk/Cargo.toml [hunk 3]
```diff
@@ -23,10 +23,23 @@ edition = "2021"
...
+
+[dev-dependencies]
```
[the remark this answers is not in the document]
Comment (kevinshahfws) [reply device/thunder_ripple_sdk/Cargo.toml]: Done

**Review by satlead — APPROVED** [review]

**Pull request merged**